float32 tension
float32 current
float32 temperature